### PR TITLE
blockdev_inc_backup_with_ignore:do incremental backup with ignore

### PR DESCRIPTION
**Pull request by aliang123**
Do incremental live backup with on-target-error "ignore"

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2003542

### CODE REVIEW

**Comment by aliang123** [comment]
PASS 1-Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_with_ignore.q35

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks.

File: qemu/tests/blockdev_inc_backup_with_ignore.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        self.generate_inc_files()
+        self.do_incremental_backup()
+        self.rebase_backup_images()
+        self.prepare_clone_vm()
+        self.verify_data_files()
+
+    def post_test(self):
+        if self.main_vm.is_alive():
+            self.main_vm.destroy()
+        #self.snapshot_image.remove()
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_with_ignore.py:102]: we can remove this  line

File: qemu/tests/blockdev_inc_backup_with_ignore.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                                    self.params["inc_node"],
+                                    **extra_options)
+        self.main_vm.monitor.cmd(cmd, arguments)
+        timeout = self.params.get("job_timeout", 600)
+        job_id = arguments.get("job-id", self._source_nodes[0])
+        get_event = job_utils.get_event_by_condition
+        event = get_event(self.main_vm, job_utils.BLOCK_JOB_ERROR_EVENT,
+                          timeout, device=job_id, action='ignore')
+        if not event:
+            self.test.fail("Backup job can't reach error after %s seconds",
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_with_ignore.py:56]: self.test.fail takes one arg

File: qemu/tests/blockdev_inc_backup_with_ignore.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                           timeout)
+        process.system(self.params['lv_extend_cmd'],
+                       ignore_status=False, shell=True)
+        job_utils.wait_until_block_job_completed(self.main_vm, job_id, timeout)
+
+    def get_image_by_tag(self, name):
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_with_ignore.py:62]: We can use the following function instead, defined in parent class
def disk_define_by_params(self, params, image_name):

File: qemu/tests/cfg/blockdev_inc_backup_with_ignore.cfg [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    # lvm setting
+    mount_path = /var/tmp/mnt
+    vg_name = vg_test
+    lv_name = lv_test
+    lv_size = 150M
+    block_raw_name = lvm_block
+    emulated_image = images/${block_raw_name}
+    emulated_image_size = ${image_size_data1}
+    lv_extend_cmd = "lvextend -L ${emulated_image_size} /dev/${vg_name}/${lv_name} && resize2fs /dev/${vg_name}/${lv_name}"
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_inc_backup_with_ignore.cfg:38]: Add the following:
force_remove_image = yes
remove_emulated_image = yes
to remove vg/pv and loop device

**Comment by aliang123** [comment]
PASS 1-Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_with_ignore.q35

File: qemu/tests/cfg/blockdev_inc_backup_with_ignore.cfg (hunk 4 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_inc_backup_with_ignore.cfg:38]: Have updated, thanks zhenchao,

File: qemu/tests/blockdev_inc_backup_with_ignore.py (hunk 3 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_with_ignore.py]: Have updated, thanks zhenchao.

File: qemu/tests/blockdev_inc_backup_with_ignore.py (hunk 2 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_with_ignore.py]: Have updated, thanks zhenchao.

File: qemu/tests/blockdev_inc_backup_with_ignore.py (hunk 1 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_with_ignore.py]: Have updated, thanks zhenchao.

**Comment by ngu-niny** [comment]
PASS 1-Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_with_ignore

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**